substrate/frame/treasury/src/lib.rs

```diff
@@ -987,6 +989,19 @@ impl<T: Config<I>, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>> for Palle
 	}
 }
 
+/// Implement the `OnUnbalanced` handler for [`frame_support::traits::fungible`] trait currency.
+pub struct FungibleCompat<T>(PhantomData<T>);
+impl<T: Config> OnUnbalanced<Credit<T::AccountId, T::Currency>> for FungibleCompat<T> {
+	fn on_nonzero_unbalanced(credit: Credit<T::AccountId, T::Currency>) {
+		use frame_support::traits::fungible::Balanced;
+		let numeric_amount = credit.peek();
+
+		let _ = T::Currency::resolve(&Pallet::<T>::account_id(), credit).defensive();
+
+		Pallet::<T>::deposit_event(Event::Deposit { value: numeric_amount });
```

Comment:
the even already publishing already there on `resolve` completion, done by implementation of `fungibles::Balanced::done_deposit` method.